components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -375,6 +351,9 @@ export default class ChannelDetails extends React.Component<ChannelDetailsProps,
                 serverError = <FormError error={resultWithError.error.message}/>;
             } else {
                 await actions.getGroups(channelID);
+                if (isPrivacyChanging) {
+                    await actions.getChannelModerations(channelID).then(this.getChannelModerationsCallback);
+                }
```

Comment:
this will be called before the patch and therefore if the user has modified channel moderated permissions while also toggling from private to public the channel moderated permissions changes will tossed and replaced with what exists on the server

Reply:
That won't do at all! I'll fix it, unless Hossein is right and you're talking care of this in another PR?